lib/jekyll/entry_filter.rb

```diff
@@ -1,27 +1,30 @@
 module Jekyll
   class EntryFilter
-    SPECIAL_LEADING_CHARACTERS = ['.', '_', '#', '~'].freeze
-
     attr_reader :site
+    SPECIAL_LEADING_CHARACTERS = [
+      '.', '_', '#', '~'
+    ].freeze
 
     def initialize(site, base_directory = nil)
       @site = site
-      @base_directory = derive_base_directory(@site, base_directory.to_s.dup)
+      @base_directory = derive_base_directory(
+        @site, base_directory.to_s.dup
+      )
     end
 
     def base_directory
       @base_directory.to_s
     end
 
     def derive_base_directory(site, base_dir)
-      if base_dir.start_with?(site.source)
-        base_dir[site.source] = ""
-      end
+      base_dir[site.source] = "" if base_dir.start_with?(site.source)
       base_dir
     end
 
     def relative_to_source(entry)
-      File.join(base_directory, entry)
+      File.join(
+        base_directory, entry
+      )
```

Comment:
Also @parkr if you want I've already got all that done, I just never pushed it up.  If you want I'll push it.